client/my-sites/media-library/proxied-image.jsx

```diff
@@ -0,0 +1,43 @@
+/**
+ * External dependencies
+ */
+import React, { useEffect, useState } from 'react';
+import { noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+
+const { Blob } = globalThis; // The linter complains if I don't do this...?
+
+export default function( { alt, mediaUrl, onLoad = noop, siteSlug, style } ) {
+	const [ imageData, setImageData ] = useState( null );
+
+	useEffect( () => {
+		if ( imageData !== null ) {
+			return;
+		}
+
+		wpcom.undocumented().getAtomicSiteMediaViaProxyRetry( siteSlug, mediaUrl, ( err, data ) => {
+			if ( ! ( data instanceof Blob ) ) {
+				setImageData( false );
+				return;
+			}
+
+			const imageUrl = URL.createObjectURL( data );
```

Comment:
Per: https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL

...we should make sure to call `URL.revokeObjectURL()` on these when the component unmounts or the memory won't be released until the document is unloaded.